### PR TITLE
Add basic scala parameterisation support

### PR DESCRIPTION
**Pull request by mpacer**
~~I still need to add tests and make sure all the data types are correctly specified, but~~ this worked on the command line with strings.

Tests have now been added & data types are tested.

### CODE REVIEW

File: papermill/preprocess.py [hunk 1]
```diff
@@ -3,8 +3,7 @@
 import datetime
 
 from concurrent import futures
-from nbconvert.preprocessors import ExecutePreprocessor
```
[the remark this answers is not in the document]
Comment (mpacer) [reply papermill/preprocess.py]: This isn't a change in nbconvert, you could have done this at least as of nbconvert 5.0.

[the remark this answers is not in the document]
Comment (mpacer) [reply papermill/preprocess.py]: sure I'll make this a separate PR.

File: papermill/execute.py [hunk 2]
```diff
@@ -283,6 +316,13 @@ def build_r_params(parameters):
         param_content += '{} = {}\n'.format(var, _translate_type_r(val))
     return param_content
 
+@register_param_builder("scala")
+def build_python_params(parameters):
+    """Writers parameter assignment code for Python kernels."""
+    param_content = "// Parameters\n"
+    for var, val in parameters.items():
+        param_content += 'val {} = {}\n'.format(var, _translate_type_scala(val))
```
[the remark this answers is not in the document]
Comment (mpacer) [reply papermill/execute.py]: I don't think that we're going to be able to handle that with the current type system that papermill implements.

could you link to what you're talking about re: play or circe?

**Comment by mpacer** [comment]
@MSeal Addressed your comments except for the one that would require the much richer type system which I think is outside of the scope of this PR.

File: papermill/execute.py [hunk 3]
```diff
@@ -257,6 +257,38 @@ def _translate_type_r(val):
...
+
+def _translate_type_dict_scala(val):
+    """Translate dicts to scala maps"""
+    escaped = ', '.join(["{} -> {}".format(_translate_type_str_r(k), _translate_type_r(v)) for k, v in val.items()])
+    return 'Map({})'.format(escaped)
+
+def _translate_type_list_scala(val):
+    """Translate list to scala."""
+    escaped = ', '.join([_translate_type_r(v) for v in val])
+    return 'List({})'.format(escaped)
```
[the remark this answers is not in the document]
Comment (mpacer) [reply papermill/execute.py]: Oh i missed this one… can you explain why?

[the remark this answers is not in the document]
Comment (mpacer) [reply papermill/execute.py]: Note: as of 807f8e7 this is fixed too.

File: papermill/execute.py [hunk 4]
```diff
@@ -283,6 +315,13 @@ def build_r_params(parameters):
         param_content += '{} = {}\n'.format(var, _translate_type_r(val))
     return param_content
 
+@register_param_builder("scala")
+def build_python_params(parameters):
```
[the remark this answers is not in the document]
Comment (mpacer) [reply papermill/execute.py]: fixed

File: papermill/execute.py [hunk 5]
```diff
@@ -257,6 +257,42 @@ def _translate_type_r(val):
...
+                         for k, v in val.items()])
+    return 'Map({})'.format(escaped)
+
+def _translate_type_list_scala(val):
+    """Translate list to scala."""
+    escaped = ', '.join([_translate_type_scala(v) for v in val])
+    return 'Seq({})'.format(escaped)
+
+def _translate_type_scala(val):
+    """Translate each of the standard json/yaml types to appropiate objects in R."""
```
Comment (MSeal) [inline papermill/execute.py]: /s/R/scala/

Comment (mpacer) [reply papermill/execute.py]: fixed with 82c2328

**Pull request merged**